### PR TITLE
Cleanup internal Eclair API

### PR DESCRIPTION
**Pull request by Christewart**
I ended up writing this to use for a suredbits project, but figured it would be better to just open source and allow for a little bit of a cleaner abstraction in eclair that hides all of the actor stuff. This currently implements a subset of the Actor messages `Service`. 

The goal with this is to replace all of the actor asks in `Service` with this eventually. For now I just wanted to get this out here and see what people think. 

I think the return types for some of the Actors (`openChannel`, `connectToNode`) probably should return something different than just a `String`.

### CODE REVIEW

**Comment by Christewart** [comment]
I think I got all of the methods that are used in `Service`. Think this is ready for merge after review

File: eclair-core/src/main/scala/fr/acinq/eclair/api/EclairClient.scala [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    sendToChannel(id, CMD_FORCECLOSE).mapTo[String]
+  }
+
+  override def channels(pubKeyOpt: Option[PublicKey]): Future[Vector[LocalChannelInfo]] = {
+    kitF.flatMap { kit =>
+
+      val channelsIdF: Future[Iterable[BinaryData]] = (kit.register ? 'channels)
+        .mapTo[Map[BinaryData, ActorRef]]
+        .map { m :Map[BinaryData, ActorRef] =>
+          if (pubKeyOpt.isDefined) {
```
Comment (Christewart) [inline eclair-core/src/main/scala/fr/acinq/eclair/api/EclairClient.scala]: Review needed here, I think this is correct but would like more eye balls on it.

File: eclair-core/src/main/scala/fr/acinq/eclair/api/EclairClient.scala [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+  private val rpcClient = {
+    new BasicBitcoinJsonRPCClient(
+      user = setup.config.getString("bitcoind.rpcuser"),
+      password = setup.config.getString("bitcoind.rpcpassword"),
+      host = setup.config.getString("bitcoind.host"),
+      port = setup.config.getInt("bitcoind.rpcport"))
+  }
+
+  def generateBlocks(n: Int): Future[Unit] = {
```
Comment (Christewart) [inline eclair-core/src/main/scala/fr/acinq/eclair/api/EclairClient.scala]: Convenient for testing purposes. Can remove

**Comment by Christewart** [comment]
Here is a PR in bitcoin-s that could be used to revive this PR eventually, specifically the EclairApi part.

**Comment by pm47** [comment]
This has been done as part of #894.

**Pull request closed without merging**